src/qml/controls/BlockClock.qml

```diff
@@ -0,0 +1,106 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+Item {
+    id: root
+
+    Layout.alignment: Qt.AlignCenter
+    width: size
+    height: size
+
+    property real progress: 0
+    property int remainingTime: 0
+
+    property int size: 200
+    property real arcBegin: 0
+    property real arcEnd: progress * 360
+    property real lineWidth: 4
+    property string colorCircle: "#f1d54a"
+    property string colorBackground: Theme.color.neutral2
+
+    property alias endAnimation: animationArcEnd.enabled
+
+    property int animationDuration: 250
+
+    onArcEndChanged: canvas.requestPaint()
+
+    Behavior on arcEnd {
+       id: animationArcEnd
+       enabled: true
+       NumberAnimation {
+            easing.type: Easing.Bezier
+            easing.bezierCurve: [0.5, 0.0, 0.2, 1, 1, 1]
+            duration: root.animationDuration
+        }
+    }
+
+    Canvas {
```

Comment:
instead of canvas, this should be the simpler ShapePath: https://doc.qt.io/qt-6/qml-qtquick-shapes-shapepath.html

Reply:
I looked through using the ShapePath to paint the BlockClock. However, using this would require us to introduce a new dependency, `qml-module-qtquick-shapes` for building the GUI.

So I think this suggestion needs further discussion before implementing it.

A friendly ping, @hebasto

What do you think about this suggestion?

Reply:
Unfortunately, from my experiments, `ShapePath` and related items have issues. The most annoying issue happens with small curves where tessellation yields bad curve smoothness. Another issue is that with complex drawings you end up with lots of items and bindings everywhere, which is also not that performant.

Going forward a stack/composition of Canvas is the best option, like photoshop layers. For instance, the Canvas on the bottom could be static or updated less often than the one on the top.